OpenRA.Mods.Common/Widgets/Logic/Editor/SaveMapLogic.cs

```diff
@@ -28,6 +28,20 @@ struct MapFileTypeInfo
 			public string UiLabel;
 		}
 
+		class SaveDirectory
+		{
+			public readonly Folder Folder;
+			public readonly string DisplayName;
+			public readonly MapClassification Classification;
+
+			public SaveDirectory(Folder folder, MapClassification classification)
+			{
+				Folder = folder;
+				DisplayName = Platform.UnresolvePath(Folder.Name);
```

Comment:
Do you have any suggestions on how to make this work without reworking the save dialog (the dropdown isn't wide enough to append a prefix), and also work in the case where mods might have multiple system or user directories?